### PR TITLE
speedreader: Improve description truncation

### PR DESCRIPTION
**Pull request by keur**
For sentences that are more than 200 characters (roughly 3 sentences)
truncate the result at the first sentence. Right now some descriptions
are getting prematurely truncated on abbreviations.

Resolves https://github.com/brave/brave-browser/issues/17983

## Submitter Checklist:

- [ ] I confirm that no security/privacy review [is needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or that I have [requested](https://github.com/brave/security/issues/new/choose) one
- [ ] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [ ] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [ ] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [ ] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [ ] Checked the PR locally: `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests`, `npm run lint`, `npm run gn_check`, `npm run tslint`
- [ ] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

### CODE REVIEW

File: components/speedreader/rust/lib/src/readability/src/nlp.rs [hunk 1]
```diff
@@ -0,0 +1,121 @@
+use regex::{Regex, RegexSet};
+use std::cmp::max;
+use std::collections::HashSet;
+
+// A list of common abbreviations
+static EN_ABBREVIATIONS_LIST: [&str; 80] = [
```
Comment (antonok-edm) [inline components/speedreader/rust/lib/src/readability/src/nlp.rs:6]: It'd be good to reference the source or methodology used to generate this list, so its completeness can be evaluated in the future

**Review by antonok-edm — APPROVED** [review]
Could be a followup or wontfix as far as I'm concerned, but I'm curious if you've put any thought into how this could work for clickbait articles where the description is something like `OMG!! You'll never believe... ` or even `Wow! ...`. I imagine we can assume a suitable sentence boundary will not occur within the first N characters, for some small N.

File: components/speedreader/rust/lib/src/readability/src/nlp.rs (hunk 1 above)
Comment (keur) [reply components/speedreader/rust/lib/src/readability/src/nlp.rs:7]: Fixed in https://github.com/brave/brave-core/pull/10027/commits/29af44a144153288982d7bb859609cdad197aeb7

**Comment by keur** [comment]
@antonok-edm That's a good point. We could add some special handling to skip a boundary that is present in a sentence of one word long. I would do this in a follow-up because I still need to think about it more.

Your case got me thinking about another case. The preferred stylization of Yahoo! and other companies could trigger a false positive. So I added another constraint that a sentence boundary is either (1) followed by a capitalized token or (2) Ends the sentence.

**Comment by keur** [comment]
speedreader has its own tests. merging.

**Pull request merged**